eox_core/api/v1/tests/integration/test_views.py

```diff
@@ -1018,6 +1076,427 @@ def test_update_valid_enrollment_using_force_flag(self, param: str) -> None:
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertNotIn("warning", response_data)
+        self.assertEqual(response_data["email"], pre_enrollment_data["email"])
+        self.assertEqual(response_data["course_id"], pre_enrollment_data["course_id"])
+        self.assertEqual(response_data["auto_enroll"], pre_enrollment_data["auto_enroll"])
+        pre_enrollment_response = self.get_pre_enrollment(self.tenant_x, data=pre_enrollment_data)
+        self.assertEqual(pre_enrollment_response.status_code, status.HTTP_200_OK)
+
+    @ddt.data(
+        {"course_id": ["This field is required."]},
+        {"email": ["This field is required."]},
+        {"course_id": ["This field is required."], "email": ["This field is required."]},
+    )
+    def test_create_pre_enrollment_missing_required_fields(self, errors: dict) -> None:
+        """
+        Test creating an pre-enrollment with missing required fields.
+
+        Open edX definitions tested:
+        - `validate_org`
+        - `get_valid_course_key`
+
+        Expected result:
+        - The status code is 400.
+        - The response contains a message about the missing fields.
+        - The pre-enrollment is not created in the tenant.
+        """
+        complete_pre_enrollment_data = {
+            "email": self.user["email"],
+            "course_id": self.demo_course_id,
+        }
+        incomplete_pre_enrollment_data = {
+            key: value for key, value in complete_pre_enrollment_data.items() if key not in errors
+        }
+
+        response = self.create_pre_enrollment(self.tenant_x, incomplete_pre_enrollment_data)
+
+        response_data = response.json()
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(response_data, errors)
+        pre_enrollment_response = self.get_pre_enrollment(self.tenant_x, data=complete_pre_enrollment_data)
+        self.assertEqual(pre_enrollment_response.status_code, status.HTTP_404_NOT_FOUND)
+
+    def test_create_pre_enrollmet_already_exists(self) -> None:
+        """
+        Test creating an pre-enrollment that already exists in a tenant.
+
+        Open edX definitions tested:
+        - `validate_org`
+        - `get_valid_course_key`
+        - `CourseEnrollmentAllowed`
+        - `get_courseware_courses`
+
+        Expected result:
+        - The status code is 400.
+        - The response contains an error message about the pre-enrollment already existing.
+        - The enrollment is not created in the tenant.
+        """
+        pre_enrollment_data = {
+            "email": self.user["email"],
+            "course_id": self.demo_course_id,
+        }
+        self.create_pre_enrollment(self.tenant_x, pre_enrollment_data)
+
+        response = self.create_pre_enrollment(self.tenant_x, pre_enrollment_data)
+        response_data = response.json()
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
```

Comment:
It's 404 because the `NotFound` class by default has this status code:
https://github.com/eduNEXT/eox-core/blob/7f13278e8e643fc2abdcdae60cbb0c4605bfe51e/eox_core/edxapp_wrapper/backends/pre_enrollment_l_v1.py#L45
Maybe using the `ValidationError` class (400) in this case is better. What do you think? Do we make the change at once?

Reply:
We should not do it here, so we don't introduce breaking changes. But let's make sure this gets reported somewhere, like in the repository issues.

Reply:
I created an issue: https://github.com/eduNEXT/eox-core/issues/291